src/ripple/app/paths/impl/StrandFlow.h

```diff
@@ -291,6 +291,9 @@ class ActiveStrands
         // Swap, don't move, so we keep the reserve in next_
         cur_.clear ();
         std::swap (cur_, next_);
+        for(auto& strand : cur_)
```

Comment:
Would be good to add a comment that the motivation for the nested `for` loop is to clear the `BookStep::becameUnfunded_` containers.  The name `restart` suggests there's something more general going on.